File: PipelineService/Models/MqttMessages/NodeExecutionRequestDoubleInput.cs
using System;
using System.Collections.Generic;

namespace PipelineService.Models.MqttMessages
{
	public class NodeExecutionRequestDoubleInput : NodeExecutionRequest
	{
		/// <summary>
		/// The first input dataset the operation will be performed on.
		/// Might be null if the is using a producing hash for identifying the input dataset.
		/// </summary>
		public Guid? InputDataSetOneId { get; set; }

		/// <summary>
		/// The hash value of the node who's output dataset is the first input for this operation.
		/// </summary>
		public string InputDataSetOneHash { get; set; }

		/// <summary>
		/// The second input dataset the operation will be performed on.
		/// Might be null if the is using a producing hash for identifying the input dataset.
		/// </summary>
		public Guid? InputDataSetTwoId { get; set; }

		/// <summary>
		/// The hash value of the node who's output dataset is the second input for this operation.
		/// </summary>
		public string InputDataSetTwoHash { get; set; }

		/// <summary>
		/// The configuration of the operation. 
		/// </summary>
		public IDictionary<string, string> OperationConfiguration { get; set; } = new Dictionary<string, string>();
	}
}

File: OperationWorker/src/models/node_execution_request_double_input.py
from src.models.node_execution_request import NodeExecutionRequest


class NodeExecutionRequestDoubleInput(NodeExecutionRequest):
    def __init__(self, deserialized: dict):
        super().__init__(deserialized)

        self.input_dataset_one_id: str = deserialized['InputDataSetOneId']
        self.input_dataset_one_hash: str = deserialized['InputDataSetOneHash']
        self.input_dataset_two_id: str = deserialized['InputDataSetTwoId']
        self.input_dataset_two_hash: str = deserialized['InputDataSetTwoHash']
        self.operation_configuration: dict = deserialized['OperationConfiguration']

    def get_input_dataset_one_id(self):
        return self.input_dataset_one_id

    def get_input_dataset_one_hash(self):
        return self.input_dataset_one_hash

    def get_input_dataset_two_id(self):
        return self.input_dataset_two_id

    def get_input_dataset_two_hash(self):
        return self.input_dataset_two_hash

    def get_operation_configuration(self):
        return self.operation_configuration
